src/MSCorEE.Desktop/IEnumUnknownExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System.Collections.Generic;
+
+    public static class IEnumUnknownExtensions
+    {
+        public static IEnumerable<T> OfType<T>(this IEnumUnknown enumerator)
+        {
+            if (enumerator != null)
```

Comment:
If `enumerator` is null, wouldn't throwing `ArgumentNullException` be more appropriate? Isn't it an error condition if someone tries to enumerate null?